### PR TITLE
python3Packages.pecan: apply the patch for SQLAlchemy 1.4.x

### LINKED ISSUE #127290: Cannot build python3Packages.pecan
**Describe the bug**
Build failed during test phase.
last 10 lines of log:
```
       > pecan/tests/test_no_thread_locals.py::TestRestController::test_custom_method_type
       > pecan/tests/test_rest.py::TestRestController::test_basic_rest
       >   /nix/store/1xji37jwmnj0dkij6jlwwsccnqpw0d33-python3.8-webtest-2.0.32/lib/python3.8/site-packages/webtest/lint.py:359: WSGIWarning: Unknown REQUEST_METHOD: 'RESET'
       >     warnings.warn(
       >
       > -- Docs: https://docs.pytest.org/en/stable/warnings.html
       > =========================== short test summary info ============================
       > FAILED pecan/tests/test_jsonify.py::TestJsonifySQLAlchemyGenericEncoder::test_result_proxy
       > FAILED pecan/tests/test_jsonify.py::TestJsonifySQLAlchemyGenericEncoder::test_row_proxy
       > ============ 2 failed, 541 passed, 1 skipped, 27 warnings in 3.13s =============
```
full log (produced by nix log /nix/store/smrzwazd7hxy58rbmkrmlb4ml0bb4kiy-python3.8-pecan-1.4.0.drv) is available at: https://gist.github.com/applePrincess/3164282320c2067d86438c07e77564a8
**To Reproduce**
1. nix build nixpkgs#python3Packages.pecan

**Expected behavior**
A successful build

**Screenshots**

**Additional context**

**Notify maintainers**
No maintainers set so CC to list of contributers instead.
@FRidh @prusnak @SuperSandro2000 @ryantm @dotlambda @Profpatsch @Mic92 @drewrisinger @adisbladis 

**Metadata**

```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.12.10, NixOS, 21.11.20210612.d51b663 (Porcupine)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.4pre20210601_5985b8b`
 - channels(root): `"nixos-20.09.4242.cee3ca37c68, nixos-unstable-21.11pre293089.1c2986bbb80"`
 - channels(applePrincess): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

```
nixpkgs git commit: `ea12a8ea1742389348950c77e75097320bd9d2dd`
Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

### PR DESCRIPTION
**Pull request by applePrincess**
###### Motivation for this change
Fixes #127290

Promote myself as a maintainer.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 127362` at 042e3ffd run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages skipped due to time constraints:</summary>
  <ul>
    <li>bareos</li>
    <li>ceph (ceph-dev)</li>
    <li>ceph-client</li>
    <li>libceph</li>
    <li>qemu_full</li>
  </ul>
</details>
<details>
  <summary>3 packages built successfully:</summary>
  <ul>
    <li>python38Packages.WSME</li>
    <li>python38Packages.pecan</li>
    <li>python39Packages.pecan</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/license-missing.md">license-missing</a>

Package is missing a license.

Near pkgs/development/python-modules/pecan/default.nix:62:3:
```
   |
62 |   meta = with lib; {
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Consider adding a comment explaining the purpose of this patch on the line preceeding.
Near pkgs/development/python-modules/pecan/default.nix:33:5:
```
   |
33 |     (fetchpatch {
   |     ^
```
</li>
  </ul>
</details>

File: pkgs/development/python-modules/pecan/default.nix [hunk 1]
```diff
@@ -28,6 +29,14 @@ buildPythonPackage rec {
     sha256 = "4b2acd6802a04b59e306d0a6ccf37701d24376f4dc044bbbafba3afdf9d3389a";
   };
 
+  patches = [
+    (fetchpatch {
+      name = "Support-SQLAlchemy-1.4x";
```
Comment (dotlambda) [inline pkgs/development/python-modules/pecan/default.nix:34]:
```suggestion
      name = "Support-SQLAlchemy-1.4x.patch";
```

**Review by dotlambda — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/pecan/default.nix (hunk 1 above)
Comment (applePrincess) [reply pkgs/development/python-modules/pecan/default.nix]: Correct extension added

File: pkgs/development/python-modules/pecan/default.nix [hunk 2]
```diff
@@ -54,5 +63,6 @@ buildPythonPackage rec {
     description = "Pecan";
     homepage = "http://www.pecanpy.org/";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pecan/default.nix:64]:
```suggestion
    homepage = "https://www.pecanpy.org/";
```

Comment (applePrincess) [reply pkgs/development/python-modules/pecan/default.nix]: homepage url changed.

File: pkgs/development/python-modules/pecan/default.nix [hunk 3]
```diff
@@ -52,7 +61,8 @@ buildPythonPackage rec {
 
   meta = with lib; {
     description = "Pecan";
```
Comment (dotlambda) [inline pkgs/development/python-modules/pecan/default.nix:63]: Something like
```suggestion
    description = "WSGI object-dispatching web framework, designed to be lean and fast";
```
might be better. You decide what fits.

Comment (applePrincess) [reply pkgs/development/python-modules/pecan/default.nix]: sounds good, changed it!

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 127362` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package marked as broken and skipped:
</summary>
<ul>
<li>bareos</li>
</ul>
</details>
<details>
<summary>
8 packages built:
</summary>
<ul>
<li>ceph (ceph-dev)</li>
<li>ceph-client</li>
<li>libceph</li>
<li>python38Packages.WSME</li>
<li>python38Packages.pecan</li>
<li>python39Packages.pecan</li>
<li>qemu_full</li>
<li>sambaFull (samba4Full)</li>
</ul>
</details>

**Pull request merged**